### DISCUSSION TITLE
I want to enable intl extension on lambda layer with CDK

### DISCUSSION THREAD

**Discussion by asato-imanishi** [brefphp/bref#1378] [Support] [answered]
I'm using bref to build Laravel + AWS Lambda app with AWS CDK instead of serverless framework.

Bref Layer Version: `arn:aws:lambda:ap-northeast-1:209497400698:layer:php-81-fpm:22`

And I want to enable intl extension on lambda layer with CDK.

With serverless framework environment, we should enable it according to below procedure. ( put `php/conf.d/php.ini` on project root.)
https://bref.sh/docs/environment/php.html#extensions-installed-but-disabled-by-default

But with CDK, I don't know how to enable it.

Note: I'm aware that other extra PHP extensions can be enabled according to below procedure.
https://github.com/brefphp/extra-php-extensions#lambda-layers-in-details

**Comment by deleugpn** [comment, marked as answer]
You need to create a file in `{PROJECT_ROOT}/php/conf.d/php.ini` where you place `extension=intl` as documented here https://bref.sh/docs/environment/php.html#extensions-installed-but-disabled-by-default

**Reply by asato-imanishi** [reply]
Thank you!

I incorrectly put `php/conf.d/php.ini` on `cdk/` directory.
It worked by putting it on laravel project's root directory ( `codebase/` ) instead of `cdk/` directory.

```
${cdk_project_root}
├── cdk
   ├── cdk.json
   ├── php/conf.d/php.ini // wrong
├── codebase (laravel's project)
   ├── php/conf.d/php.ini // correct
```

**Comment by mnapoli** [comment]
@deleugpn already answered your question but I wanted to let you know about https://github.com/brefphp/constructs

It's still in beta, but it might be useful eventually.

**Reply by asato-imanishi** [reply]
Thank you for let me know about it!
That's useful for knowing how to setup project with cdk and bref.

**Comment by nicodemuz** [comment]
@mnapoli I have a similar issue where I'm getting the following warning:
```Warning: PHP Startup: Unable to load dynamic library 'intl' (tried: /opt/lib/php/extensions/no-debug-non-zts-20210902/intl (/opt/lib/php/extensions/no-debug-non-zts-20210902/intl: cannot open shared object file: No such file or directory), /opt/lib/php/extensions/no-debug-non-zts-20210902/intl.so (/opt/lib/php/extensions/no-debug-non-zts-20210902/intl.so: cannot open shared object file: No such file or directory)) in Unknown on line 0```

Any idea where I'm going wrong? I'm using docker images with the following Dockerfile:
```
FROM bref/php-81-fpm

# Install PHP Imagick
COPY --from=bref/extra-imagick-php-81:0.12.1 /opt /opt

# Install fonts and font building tools
RUN yum -y install stix-fonts fontconfig wget tar xz make fontforge

# Build Thai fonts
RUN wget https://linux.thai.net/pub/thailinux/software/fonts-tlwg/fonts-tlwg-0.6.1.tar.xz -O /tmp/fonts-tlwg-0.6.1.tar.xz; \
    cd /tmp; \
    tar -xf fonts-tlwg-0.6.1.tar.xz; \
    cd fonts-tlwg-0.6.1; \
    ./configure; \
    make; \
    make install; \
    cd ..; \
    mkdir thai; \
    find . -name '*.ttf' | cpio -pdm ./thai; \
    cd thai; \
    cp -R fonts-tlwg-0.6.1/ /usr/share/fonts/; \
    fc-cache -frv /usr/share/fonts/

# Install wkhtmltopdf
RUN yum -y install openssl-devel bzip2-devel libffi-devel wget tar gzip make gcc-c++
RUN wget https://github.com/wkhtmltopdf/packaging/releases/download/0.12.6-1/wkhtmltox-0.12.6-1.amazonlinux2.x86_64.rpm
RUN yum -y install wkhtmltox-0.12.6-1.amazonlinux2.x86_64.rpm

# Copy Symfony codebase
COPY . /var/task

# Configure the handler file (the entrypoint that receives all HTTP requests)
CMD ["public/index.php"]
```

**Reply by mnapoli** [reply]
@nicodemuz could you please open a separate discussion?

FYI it seems that you are using incompatible versions of bref images (no version = v2) and bref-extra (0.x is for bref v1). But let's discuss that in a new thread because this is not related to AWS CDK.

**Reply by nicodemuz** [reply]
@mnapoli Thank you. I created a new thread: https://github.com/brefphp/bref/discussions/1525
